src/routing_table/mod.rs

```diff
@@ -909,7 +909,7 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         }
         for (prefix, group) in &self.groups {
             if has_enough_nodes && group.len() < self.min_group_size {
-                if group.is_empty() && allow_empty_groups {
+                if group.len() <= 1 && allow_small_groups {
```

Comment:
Is it guaranteed that that it'll never be 2?
Maybe we just shouldn't do the size check at all if `allow_small_groups`?
```rust
if !allow_small_groups && has_enough_nodes && group.len() < self.min_group_size {
```